src/components/AppLayout/index.tsx

```diff
@@ -91,6 +93,14 @@ const Layout: React.FC<Props> = ({
 
   const closeMobileNotSupported = () => setMobileNotSupportedClosed(true)
 
+  const settingsBaseRoute = safeAddress
+    ? generatePath(SAFE_ROUTES.SETTINGS_BASE_ROUTE, {
```

Comment:
Instead of generating the paths and testing, you can use react-routers `matchPath` function to do the matching for you:
```ts
const hasFooter = [SAFE_ROUTES.SETTINGS_BASE_ROUTE, WELCOME_ADDRESS].some((path) => matchPath(pathname, {path}))
```